test/e2e/gitopsaddon/scripts/setup_env.sh

```
@@ -173,6 +173,59 @@ if ! ${KUBECTL} --context ${HUB_CTX} -n openshift-gitops wait --for=condition=Re
+echo ">> Waiting for ArgoCD agent principal pod"
+for i in $(seq 1 36); do
+  POD_COUNT=$(${KUBECTL} --context ${HUB_CTX} get pods -n openshift-gitops \
+    -l app.kubernetes.io/name=openshift-gitops-agent-principal --no-headers 2>/dev/null | wc -l)
+  if [ "$POD_COUNT" -gt 0 ]; then
+    echo "  Principal pod found, waiting for Ready..."
+    break
+  fi
+  echo "  Waiting for principal pod to appear... ($i/36)"
+  sleep 10
+done
+${KUBECTL} --context ${HUB_CTX} -n openshift-gitops wait \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)